File: 5to_Semestre/Algoritmos Avanzados/Concurso VJudge/c.py
def main():
    flag = False

    input_string = input()

    cnt=0
    i=0

    while(input_string[i]!='\0' and i < (len(input_string)-1)):
        if(input_string[i]=='1'):
            while(input_string[i]=='1'):
                cnt+=1
                i+=1
        else:
            while(input_string[i]=='0'):
                cnt+=1
                i+=1
        if(cnt>=7):
            flag=True
            break
        cnt=0

    if(flag):
        print("YES")
    else:
        print("NO")
    


if __name__ == "__main__":
    main()

File: 5to_Semestre/Algoritmos Avanzados/Concurso VJudge/c.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    char a[105];int cnt=0,i=0;bool flag=false;
    cin>>a;
    while(a[i]!='\0'){
        if(a[i]=='1'){while(a[i]=='1'){cnt++;i++;}}
        else{while(a[i]=='0'){cnt++;i++;}}
        if(cnt>=7){flag=true;break;}
        cnt=0;
    }
    if(flag)cout<<"YES"<<endl;
    else cout<<"NO"<<endl;
    return 0;
}